core/src/avm1/globals/text_format.rs

```diff
@@ -535,7 +551,10 @@ pub fn constructor<'gc>(
     this: Object<'gc>,
     args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error<'gc>> {
-    fn get_arg_as_i32<'gc>(activation: &mut Activation<'_, 'gc>, arg: Option<&Value<'gc>>) -> Result<Option<f64>, Error<'gc>> {
+    fn get_arg_as_i32<'gc>(
```

Comment:
so to be sure, the behavior only difference for setters, not the constructor?

Reply:
That's right. Flash had a somewhat consistent behavior before SWF8, and then changed it, but only for setters.